docs/type_traits.adoc

```diff
@@ -155,6 +155,26 @@ NOTE: Detecting structurality of a type is not yet possible in the general case,
 so there are certain structural types for which this trait will be `false`. In
 practice those types should be rare, and there should be no false positives.
 
+=== `nth_t`
+
+`nth_t` is a type alias that extracts the nth element from a pack of types:
```

Comment:
I like this utility...but I'm wondering if "nth" implies it starts at 1 rather than 0. Might be a good idea to make it explicit in the documentation.